src/common/hooks.js

```diff
@@ -89,6 +93,71 @@ export function useLocalStorage(key, initialValue) {
   return [storedValue, setValue];
 }
 
+/**
+ * Same as useLocalStrage but updates state when local storage changes
+ */
+
+export function useWatchedLocalStorage(key, initialValue) {
```

Comment:
We might be able to replace it but that would be outside the scope of this PR I'm not sure what would break if it is expecting the existing the behavior.